src/main/java/com/google/sps/servlets/GmailServlet.java

```diff
@@ -44,22 +40,21 @@
    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // get Google credential object. Ensure it is valid - otherwise return an error to client
-    Credential googleCredential = AuthenticationUtility.getGoogleCredential(request);
-    if (googleCredential == null) {
-      response.sendError(403, AuthenticationUtility.ERROR_403);
-      return;
-    }
+    // Load and verify google credential
+    super.doGet(request, response);
 
-    // Get messageIds from Gmail
-    Gmail gmailService = GmailUtility.getGmailService(googleCredential);
-    List<Message> messages = GmailUtility.listUserMessages(gmailService, "");
+    // Credential is null if user is not authenticated.
+    if (googleCredential != null) {
```

Comment:
Just curious - do you want to return an error of some sort to the client if the credential is null like before? I think it would be nice since you're already checking for it.

Reply:
Agreed - done!